### PR TITLE
(web-components) Use `ElementInternals` for Button elements

### PR DESCRIPTION
**Pull request by radium-v**
## Previous Behavior

Currently, form-associated custom elements use the `FormAssociated` class mixin. This mixin is heavy and duplicative, being copied wholesale for every component which utilizes it. While `FormAssociated` includes minimal support for [`ElementInternals`](https://developer.mozilla.org/en-US/docs/Web/API/ElementInternals), the implementation isn't fully baked.

## New Behavior

Improvements:
* Switches the `<fluent-button>` custom element to use `ElementInternals` by default. An optional polyfill may be added in the near future, as more components transition to use `ElementInternals`.
* Removes the checkbox-like implementation from `<fluent-toggle-button>` in favor of leaning on `aria-pressed` (an element with a `role` of `button` and the `aria-pressed` attribute is treated as a toggle button).
* Adds Playwright tests for the Button element.
* Synchronizes the `<fluent-menu-button>` and `<fluent-compound-button>` elements to work with the changes in `<fluent-button>`, since their classes are extensions of `Button`.
* Copies the original button styles into the stylesheet for `<fluent-anchor-button>`. The `AnchorButton` class isn't actually extended from `Button` but it did use the same stylesheet. Its template structure hasn't changed.
* Adds a story for `submit`- and `reset`-type buttons to the Button story file.

Other fixes:
* The Playwright config had a few minor misconfigurations:
  * The `projects` field was set to run all tests in Chromium three times in a row.
  * The viewport width and height values were flipped.

### CODE REVIEW

**Comment by Hotell** [comment]
**Some workflow notes:**

bumping deps and adding prodcution changes should be ideally done in N separate PRs as it's unrelated. 

this would be more likely a blocker if this PR would be against master branch but for you case I leave that decision up to you as it doesn't effect daily monorepo contributions.

File: azure-pipelines-pr.yml [hunk 1]
```diff
@@ -29,6 +29,9 @@ jobs:
           filePath: yarn-ci.sh
         displayName: yarn (install packages)
 
+      - script: yarn playwright install --with-deps
```
Comment (Hotell) [inline azure-pipelines-pr.yml:32]: **FYI:** 

this is ok to live in wc3 branch. 

if this would be against master we would most likely not accept this, as the cost of prolonging pipeline just for sake of  1 package is not worth the slowdown. If whole monorepo moves to use playwright for all e2e in feature then it's totally valid. 

For this use-case we would need to make this on-demand as we do within fluent-contrib https://github.com/microsoft/fluentui-contrib/blob/main/packages/nx-plugin/src/executors/playwright/executor.ts#L51-L54

File: packages/web-components/api-extractor.json [hunk 2]
```diff
@@ -18,11 +18,6 @@
     "enabled": true
   },
   "compiler": {
-    "tsconfigFilePath": "./tsconfig.lib.json",
-    "overrideTsconfig": {
-      "compilerOptions": {
-        "paths": {}
-      }
-    }
+    "tsconfigFilePath": "./tsconfig.api-extractor.json"
```
Comment (Hotell) [inline packages/web-components/api-extractor.json:21]: why is this change necessary ? another exotic tsconfig === more confusion / not following repo patterns

**Review by Hotell — APPROVED** [review]
approving as it's for wc3 branch. these changes would not be acceptable for master - added more context in comments for better understanding. ty

File: packages/web-components/api-extractor.json (hunk 2 above)
Comment (radium-v) [reply packages/web-components/api-extractor.json:21]: The API Extractor doesn't use the `tsconfigFilePath` if any `overrideTsconfig` fields are present. This was causing warnings "Subsequent property declarations must have the same type" (TS2717) with `@types/web` present.

File: azure-pipelines-pr.yml (hunk 1 above)
Comment (radium-v) [reply azure-pipelines-pr.yml:32]: How are the Playwright binaries currently made available for existing packages which use Storywright?

File: packages/web-components/src/toggle-button/toggle-button.ts [hunk 3]
```diff
@@ -1,108 +1,74 @@
...
+   *
    * @internal
    */
-  protected handleToggleButtonClick = (e: MouseEvent): void => {
-    if (!this.disabled && !this.disabledFocusable) {
-      this.checked = !this.checked;
-    }
-  };
+  private setPressedState(): void {
+    const ariaPressed = `${this.indeterminate ? 'mixed' : !!this.pressed}`;
```
Comment (chrisdholt) [inline packages/web-components/src/toggle-button/toggle-button.ts:70]: Do toggle buttons support mixed? I think that would simply be for `checkbox`. I don't think `switch` would be mixed either. https://www.w3.org/WAI/ARIA/apg/patterns/button/

I'm not sure how you'd have a mixed state for on/off. I know that it's supported in "pressed" but I am not sure it makes sense in the pattern.

**Review by chrisdholt — APPROVED** [review]

File: packages/web-components/src/toggle-button/toggle-button.ts (hunk 3 above)
Comment (radium-v) [reply packages/web-components/src/toggle-button/toggle-button.ts:70]: From MDN:

> Adding `aria-pressed` to an element with a role of [`button`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) turns the button into a toggle button. The `aria-pressed` attribute is only relevant for toggle buttons. It represents the button's current "pressed" state.
> 
> The value is "tri-state", meaning the value can be set to `true`, `false`, `mixed`, or `undefined` values. In the case of `aria-pressed`, like most tri-state value types, the default value is `undefined`.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-pressed

I can remove the `indeterminate`/`mixed` state if it isn't relevant for this particular component - I don't think we have any stylistic differences for a mixed-state toggle button.

Comment (chrisdholt) [reply packages/web-components/src/toggle-button/toggle-button.ts:70]: Let's leave it for now given this documentation - I'm not sure it infers that there should be a visual difference here, rather it pertains to the value of what this represents. Good to leave for now.
